scalafix-core/shared/src/main/scala/scalafix/internal/util/LazySymbolTable.scala

```diff
@@ -0,0 +1,113 @@
+package scalafix.internal.util
+
+import scala.meta.internal.semanticdb3.Scala.Symbols
+import java.io.InputStream
+import java.net.URI
+import java.nio.file.FileSystem
+import java.nio.file.FileSystemAlreadyExistsException
+import java.nio.file.FileSystems
+import java.nio.file.Files
+import org.langmeta.internal.io.PathIO
+import org.langmeta.io.AbsolutePath
+import org.langmeta.io.Classpath
+import scala.collection.concurrent.TrieMap
+import scala.meta.internal.semanticdb3.Scala._
+import scala.meta.internal.{semanticdb3 => s}
+
+/**
+  * Implementation of SymbolTable that lazily loads symbols on demand.
+  *
+  * @param mclasspath The classpath to load symbols from. Needs to be pre-processed by metacp,
+  *                   see ClasspathOps.toMclasspath in scalafix-cli.
+  */
+class LazySymbolTable(mclasspath: Classpath) extends SymbolTable {
+
+  // Map from symbols to the paths to their corresponding semanticdb file where they are stored.
+  private val unloadedSymbols = TrieMap.empty[String, AbsolutePath]
+  private val loadedSymbols = TrieMap.empty[String, s.SymbolInformation]
+  private val semanticdb = "META-INF/semanticdb"
+  private val semanticIdx = "META-INF/semanticdb.semanticidx"
```

Comment:
would this be a good place to add
```
private val manifest = "META-INF/MANIFEST.MF"
```
in case there is a classpath manifest in a jar?

Reply:
I think that should be handled elsewhere, for example `scala.meta.Classpath` should be able to handle.